hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsFastpathConnection.java

```diff
@@ -0,0 +1,211 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.azurebfs.services;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.thirdparty.com.google.common.annotations.VisibleForTesting;
+
+import com.microsoft.fastpath.FastpathConnection;
+import com.microsoft.fastpath.exceptions.FastpathException;
+import com.microsoft.fastpath.requestParameters.AccessTokenType;
+import com.microsoft.fastpath.requestParameters.FastpathCloseRequestParams;
+import com.microsoft.fastpath.requestParameters.FastpathOpenRequestParams;
+import com.microsoft.fastpath.requestParameters.FastpathReadRequestParams;
+import com.microsoft.fastpath.responseProviders.FastpathCloseResponse;
+import com.microsoft.fastpath.responseProviders.FastpathOpenResponse;
+import com.microsoft.fastpath.responseProviders.FastpathReadResponse;
+import com.microsoft.fastpath.responseProviders.FastpathResponse;
+import static org.apache.hadoop.fs.azurebfs.constants.AbfsHttpConstants.DEFAULT_TIMEOUT;
+import static org.apache.hadoop.fs.azurebfs.constants.HttpHeaderConfigurations.X_MS_CLIENT_REQUEST_ID;
+import static org.apache.hadoop.fs.azurebfs.services.AuthType.OAuth;
+
+/**
+ * Represents a Fastpath operation.
+ */
+public class AbfsFastpathConnection extends AbfsHttpOperation {
+
+  int defaultTimeout = Integer.valueOf(DEFAULT_TIMEOUT);
+  String fastpathFileHandle;
+  FastpathResponse response = null;
+
+  public String getFastpathFileHandle() {
+    return fastpathFileHandle;
+  }
+
+  public AbfsFastpathConnection(final AbfsRestOperationType opType,
+      final URL url,
+      final String method,
+      final AuthType authType,
+      final String authToken,
+      List<AbfsHttpHeader> requestHeaders,
+      final String fastpathFileHandle) throws IOException {
+    super(opType, url, method, authType, authToken, requestHeaders);
+    this.authType = authType;
+    this.authToken = authToken;
+    this.fastpathFileHandle = fastpathFileHandle;
+    this.requestHeaders = requestHeaders;
+  }
+
+  public String getResponseHeader(String httpHeader) {
+    return response.getResponseHeaders().get(httpHeader);
+  }
+
+  public Map<String, List<String>> getRequestHeaders() {
+    Map<String, List<String>> headers
+        = new HashMap<String, java.util.List<String>>();
```

Comment:
minor comments for this file:
java.util.List --> List
constructor signature: any reason why requestHeaders not final?
getRequestHeader() method: can we use use const EMPTY_STRING